programs/drift/src/instructions/user.rs

```diff
@@ -653,6 +653,11 @@ pub fn handle_deposit<'c: 'info, 'info>(
     } else {
         DepositExplanation::None
     };
+    let signer = if ctx.accounts.authority.key() == user.authority {
+        Some(ctx.accounts.authority.key())
+    } else {
+        None
```

Comment:
not sure if it changes anything, but should this just always be ctx.accounts.authority? i guess it'd make the event larger